src/dotnet/Program.cs

```diff
@@ -253,6 +255,27 @@ private static bool IsDotnetBeingInvokedFromNativeInstaller(TopLevelCommandParse
             }
         }
 
+        internal static void InitializeEnvironment()
+            => InitializeEnvironment(new EnvironmentProvider());
+
+        internal static void InitializeEnvironment(IEnvironmentProvider env)
+        {
+            var dotnetRootEnv = env.Is64BitProcess
+                ? "DOTNET_ROOT"
+                : "DOTNET_ROOT(x86)";
+
+            try
+            {
+                var rootPath = Path.GetDirectoryName(new Muxer().MuxerPath);
+                env.SetEnvironmentVariable(dotnetRootEnv, rootPath, EnvironmentVariableTarget.Process);
+            }
+            catch
+            {
+                // swallow and ignore in the event MuxerPath throws because it cannot find dotnet.exe
+                return;
```

Comment:
Can we at least catch `Exception` and call `ReportAsWarning` (extension method) on it to log when verbose output is enabled?